functions/src/index.ts

```diff
@@ -48,28 +48,24 @@ export const getHintKit = onFlow(
   }
 );
 
-export const resetGame = onDocumentUpdated(
-  "boardInfo/solvedWordsCount",
+export const resetGame = onDocumentCreated(
+  "solvedWords/{docId}",
   async (event) => {
     try {
-      const data = event.data?.after.data();
-      const previousData = event.data?.before.data();
-
-      if (data?.value == previousData?.value) {
-        return null;
-      }
-
       const db = getFirestore();
 
+      const solvedWordsCollection = db.collection("solvedWords");
+      const solvedWordsSnapshot = await solvedWordsCollection.get();
+      const solvedWordsCount = solvedWordsSnapshot.size;
```

Comment:
Does this create the collection if it's not available? When a collection is empty, it's automatically removed, so we have to create it again.